ocaml/testsuite/tests/typing-layouts-float64/alloc.ml

```diff
@@ -0,0 +1,90 @@
+(* TEST
+   flags = "-extension layouts_alpha"
+   * native
+*)
+
+(* A test comparing allocations with unboxed floats to allocations with boxed
+   floats. *)
+
+(* CR layouts v2: test shouldn't print exact allocation numbers. And I've only
+   run it on closure - surely different on other middle ends. *)
+
+
+module type Float_u = sig
+  external to_float : float# -> (float[@local_opt]) = "%box_float"
+  external of_float : (float[@local_opt]) -> float# = "%unbox_float"
+
+  val ( + ) : float# -> float# -> float#
+  val ( - ) : float# -> float# -> float#
+  val ( * ) : float# -> float# -> float#
+  val ( / ) : float# -> float# -> float#
+  val ( ** ) : float# -> float# -> float#
+  val ( > ) : float# -> float# -> bool
+end
+
+module Float_u : Float_u = struct
+  external to_float : float# -> (float[@local_opt]) = "%box_float"
```

Comment:
This should get a double-check from a locals person. I think the other `[@local_opt]` functions are parametrically locality-polymorphic, like `id`. That is, they work either on local types or global types, but the choice doesn't affect runtime behavior. This here is different, in that the runtime behavior depends on the choice of locality. This *might* work, given that the implementation of the primitive takes the mode. But I wonder whether this implementation approach breaks new ground -- hence my desire for further review.

Reply:
Yeah - to be clear, I just stole these from [#1117](https://github.com/ocaml-flambda/flambda-backend/pull/1117), we should definitely get another set of eyes on them.